### PR TITLE
update coding style and test facilities

### PR DESCRIPTION
**Pull request by baentsch**
Addresses increasing problems with different `clang-format` versions (e.g., #473, #476); adopts mechanism from https://github.com/open-quantum-safe/liboqs/pull/1861 (kudos to @SWilson4 ).

Anyone feeling strongly about the previous (more OpenSSL-oriented, but not totally so thoroughly tested) and the new (LLVM with OpenSSL's 4 indents), please speak up. Both options LGTM -- but the new variant seems to work more consistently across `clang-format` versions).

CI for coding style also already upgraded in preparation for a fix to #473 (and a general update to more current platforms).

### CODE REVIEW

File: scripts/do_code_format.sh
Comment (SWilson4) [inline scripts/do_code_format.sh:23]:
```suggestion
   echo "To fix, consider running \"find oqsprov test -name '*.[chS]' | xargs sed -i 's/\r//' \"."
```
Assuming that "tests" should be "test" as in the commands above.

**Review by SWilson4 — APPROVED** [review]

**Review by thb-sb — APPROVED** [review]

**Comment by baentsch** [comment]
@ryjones What rights do I need to look at and change PR merge rules to overcome the merge blocker in this PR? Two approvals incl. one from the person who did the original style-check code (@thb-sb ) are more than enough for this PR -- but I don't have permissions any more to do anything about this:
image


Who can grant these rights? Why do they need to be kept away from me? Behind this the by now "usual" reminder: Please consider [answering the underlying question which rights a maintainer in LF needs (to get)](https://github.com/open-quantum-safe/tsc/issues/45)? If answered, these problems would be resolved before they hit. It's really not funny (motivating, etc.) to keep banging into permissions limitations.

@hartm @dstebila fya as you initiated this. fwiw, I'll be mostly offline for the next week, but it'd be nice to find some answers when I'm back.

**Comment by SWilson4** [comment]
It looks to me like it's due to "Require Review from Codeowners" (see Additional Setting under "Require a Pull Request Before Merging" [here](https://github.com/open-quantum-safe/oqs-provider/rules/617906?ref=refs%2Fheads%2Fmain)). Previously this would have been moot, but https://github.com/open-quantum-safe/oqs-provider/pull/458 added codeowners who now are required to approve things. Presumably the automation doesn't detect that the changes to, say, `oqs_sig.c`, which has @feventura as a codeowner, are trivial.

**Comment by ryjones** [comment]
OK I found the setting and turned it off. The issue was there were no CODEOWNERs set.

**Comment by ryjones** [comment]
This is a consequence of a change you made on 07 APR. The audit log shows:

Screenshot 2024-08-09 at 12 00 37

As @SWilson4 notes, the commit you made to add a CODEOWNERS resulted in your inability to merge.

**Comment by baentsch** [comment]
> This is a consequence of a change you made on 07 APR.

That I knew (and is obvious as I had the required GH permissions to do so at the time); the problem is that you removed my GH permissions (e.g., to update the rule set) since.

Hence, my question above was how can I keep maintaining the rule set (and also the project which I created)? Which GH permissions do I need? Or is that one of the things [a LF maintainer should _not_ be able to do](https://github.com/open-quantum-safe/tsc/issues/45)?

**Comment by ryjones** [comment]
[Looking into it](https://github.com/open-quantum-safe/tsc/pull/62) I updated this PR

**Pull request merged**
